app/main/views/suppliers.py

```diff
@@ -242,27 +243,84 @@ def edit_supplier_organisation_size():
     form = CompanyOrganisationSizeForm()
 
     if request.method == 'POST':
+        api_error = None
         if form.validate_on_submit():
-            data_api_client.update_supplier(supplier_id=current_user.supplier_id,
-                                            supplier={"organisationSize": form.organisation_size.data},
-                                            user=current_user.email_address)
-            return redirect(url_for('.supplier_details'))
+            try:
+                data_api_client.update_supplier(supplier_id=current_user.supplier_id,
+                                                supplier={"organisationSize": form.organisation_size.data},
+                                                user=current_user.email_address)
 
-        current_app.logger.warning(
-            "supplieredit.fail: organisation-size:{osize}, errors:{osize_errors}",
-            extra={
-                'osize': form.organisation_size.data,
-                'osize_errors': ",".join(form.organisation_size.errors)
-            })
+            except APIError as e:
+                api_error = e.message
+                current_app.logger.warning(f"supplieredit.fail: api error:{api_error}")
```

Comment:
Does this need the `extra` to set the api_error value?

Reply:
Nah, it's a formatted string.

Reply:
Ah, but I think something to do with the way our JSON logging works might need it done with the "extra" fields, and that's why it's done that way everywhere else we log stuff.

Reply:
Well, it's gone now anyway. 👍